mcp-server-open-api-spec/README.md

```diff
@@ -0,0 +1,223 @@
+# Custom Open API Spec MCP Server on Databricks Apps
+
+This repository provides a framework for creating [Custom Model Context Protocol (MCP) servers](https://docs.databricks.com/aws/en/generative-ai/mcp/custom-mcp) using [Databricks Apps](https://www.databricks.com/product/databricks-apps). It enables customers to create MCP servers that connect to external REST APIs defined by [OpenAPI specifications](https://github.com/OAI/OpenAPI-Specification), with secure authentication through [Databricks external connections](https://docs.databricks.com/aws/en/generative-ai/agent-framework/external-connection-tools).
+
+## Overview
+
+This MCP server acts as a bridge between LLM agents and external REST APIs. It:
+
+1. **Loads OpenAPI specifications** from the specified UC Volume Path
+2. **Provides three main tools** to LLM agents:
+   - `list_api_endpoints` - Discover available API endpoints
+   - `get_api_endpoint_schema` - Get detailed schema for specific endpoints
+   - `invoke_api_endpoint` - Execute API calls with proper authentication
+3. **Uses Databricks UC external connections** for secure authentication to external services
+4. **Runs on Databricks Apps** for scalable hosting
+
+## Quick Start
+
+### 1. Configure Your External API
+
+Upload your OpenAPI Spec (`spec.json`) to your desired Volume. Configure this volume path as a dependent resource in your Databricks App
+
+### 2. Create UC External Connection
+
+Create a Unity Catalog external connection for authentication to your external service:
+
+```sql
+-- Example: Create connection with API key authentication
+CREATE CONNECTION your_api_connection
+  TYPE http
+  OPTIONS (
+    'host' = 'https://your-api.example.com',
+    'authentication' = 'API_KEY',
+    'api_key' = secret('your-scope', 'your-api-key')
```

Comment:
This is not valid. Can we provide a valid bearer_token or even better u2m example?